ZBED4.py

```diff
@@ -0,0 +1,257 @@
+        if self.is_type():
+            return len(self.seq)
+    
+    def seq_to_print(self) -> str:
+        """Removes \t, \n and " " in sequence."""
+        return self.seq.replace('\n', '').replace('\t', '').replace(" ", '').upper()
+
+    def get_subseq(self, start:int, stop=None) -> list:
+        """Make slice of sequence or return one letter if stop is None as default.
+        """
+        if self.is_type():
+            if stop is None:
+                return self.seq[start]
+            return self.seq[start:stop:1]
+    
+    def is_type(self) -> bool:
+        """Checks the correctness of the correspondence between type and sequence."""
+        if self.type == 'DNA':
+            return set(self.seq.upper()) <= self.__DNA_ALPHABET
+        elif self.type == 'RNA':
+            return set(self.seq.upper()) <= self.__RNA_ALPHABET
+        elif self.type == 'protein':
+            return set(self.seq.upper()) <= self.__PROTEIN_ALPHABET
+        elif self.type not in ('DNA', 'RNA', 'protein'):
+            raise ValueError(f'Unknown type ({self.type}) of biological sequence!')    
+
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    """
+    A child class of "BiologicalSequence" that implements work with nucleic acids.
+
+    Methods:
+    + gc_content(self) : count G and C in sequence and return result in %
+    + complement(self) : find compliment sequence
+
+    NOTE: correct sequence maens that is_type(self) is True
+    """
+    def __init__(self, seq:str, type:str) -> None:
+        """
+        RNA_COMPLEMENT_NUCLS (dict): compliment nucliotides of RNA in upper and lower cases
+        DNA_COMPLEMENT_NUCLS (dict): compliment nucliotides of DNA in upper and lower cases
+        """
+        super().__init__(seq, type)
+        self.__RNA_COMPLEMENT_NUCLS = { "A":"U", "U":"A", "a":"u", "u":"a", "G":"C", "C":"G", "c":"g", "g":"c"}
+        self.__DNA_COMPLEMENT_NUCLS = {"A":"T", "T":"A", "a": "t", "t":"a", "G":"C", "C":"G", "g":"c", "c":"g"}
+
+    def gc_content(self) -> float:
+        """Count G and C in sequence and return result in %."""
+        if self.is_type():
+            gc_amount = 0
+            for nucl in self.seq:
+                if nucl == "G" or nucl == "C":
+                    gc_amount += 1
+            
+            cur_gc_level = gc_amount / self.seq.__len__() * 100
+            return cur_gc_level
+        
+        raise ValueError(f"Sequence doesn't match the current type ({self.type}).")
+    
+    def complement(self) -> str:
+        """Find compliment sequence."""
+        if self.is_type():
+            compl_nucls_map = {}
+            if self.type == 'DNA':
+                compl_nucls_map = self.__DNA_COMPLEMENT_NUCLS
+            elif self.type == 'RNA':
+                compl_nucls_map = self.__RNA_COMPLEMENT_NUCLS
+                
+            trans_seq = ""
+            for nucl in self.seq:
+                trans_seq += compl_nucls_map[nucl]
+            return trans_seq
```

Comment:
Опять же, тут можно избежать `if-else` истории, например следующим образом:
```suggestion
    def complement(self):
        complement_seq = ''.join(type(self).COMPLEMENT_DICT.get(base) base in self.sequence)
        return type(self)(complement_seq)
```
При этом атрибут `COMPLEMENT_DICT` определен в соответствующем (ДНК или РНК) классе. Сама логика в этом случае также (потенциально) позволит реализовывать метод `complement` для других дочерних от `NucleicAcidSequence` классов без изменения кода родительского класса